assets/components/payPalExpressButton/payPalExpressButton.jsx

```diff
@@ -2,43 +2,44 @@
 
 // ----- Imports ----- //
 
+import ReactDOM from 'react-dom';
 import React from 'react';
-import { connect } from 'react-redux';
-
-import {
-  setupPayPalExpressCheckout,
-} from 'helpers/payPalExpressCheckout/payPalExpressCheckoutActions';
-
+import type { Csrf as CsrfState } from 'helpers/csrf/csrfReducer';
+import { loadPayPalExpress, setup } from 'helpers/payPalExpressCheckout/payPalExpressCheckout';
+import type { Currency } from '../../helpers/internationalisation/currency';
 
 // ---- Types ----- //
 
 type PropTypes = {
-  setupPayPalCheckout: Function,
+  amount: number,
+  currency: Currency,
+  csrf: CsrfState,
   callback: Function,
+  setHasLoaded: Function,
+  hasLoaded: boolean,
 };
 
-
 // ----- Component ----- //
-
-const PayPalExpressButton = (props: PropTypes) => {
-  props.setupPayPalCheckout(props.callback);
-  return <div id="component-paypal-button-checkout" className="component-paypal-button-checkout" />;
-};
-
-
-// ----- Map State/Props ----- //
-
-function mapDispatchToProps(dispatch) {
-
-  return {
-    setupPayPalCheckout: (callback: Function) => {
-      dispatch(setupPayPalExpressCheckout(callback));
-    },
-  };
-
+/* eslint-disable react/prefer-stateless-function */
+class PayPalExpressButton extends React.Component<PropTypes> {
+  render() {
+    if (!this.props.hasLoaded) {
```

Comment:
Is it possible to transform this stateful component into a stateless one? As far I understand you are using a stateful component to store if the component has loaded or not. Is it possible to achieve the same behaviour by asking if the property `window.paypal` is defined?

Reply:
At the moment using state here seems to be the cleanest way to get this to work correctly. Running the action is required here, because without it we fail to re-render the component when the Promise completes. Myself and @svillafe did briefly look into other ways to do this, but none of them seemed particularly clean...